lib/utils.js

```diff
@@ -12,6 +12,31 @@ function filterPseudoHeaders (headers) {
   return dest
 }
 
+// http required the header to be encoded with latin1
+// undici will convert the latin1 buffer back to utf8
+// https://github.com/nodejs/undici/blob/2b260c997ad4efe4ed2064b264b4b546a59e7a67/lib/core/util.js#L216-L229
+// after chaining, the header will converted in wrong byte
+// Buffer.from('', 'latin1').toString('utf8') applied
+//
+// in order to presist the encoding, always encode it
+// back to latin1
+function patchUndiciHeaders (headers) {
```

Comment:
should there be an option in `undici` to leave headers as `latin1` encoding?

Reply:
I think yes. Otherwise, the `patch` is prone to error.

`undici` always trans-code the header. Maybe it should really detect the encoding before trans-code. That means the behavior should be identical to `http.request`.

Actually, I already open a issue on `undici`.
See https://github.com/nodejs/undici/issues/1903